functions/metrics.py

```diff
@@ -222,4 +241,267 @@ def intersection_over_union(polygon1_shape, polygon2_shape):
+            additional_columns (list, optional): additional columns to write in the output file. Defaults to [].
+            drop_year (bool, optional): whether to drop the year column from the labels and detections. Defaults to False.
+            tagged_results_filename (str, optional): filename for the tagged results. Defaults to 'tagged_detections'.
+            reliability_diagram_filename (str, optional): filename for the reliability diagram. Defaults to 'relability_diagram'.
+            global_metrics_filename (str, optional): filename for the global metrics. Defaults to 'global_metrics'.
+            by_class (bool, optional): whether to calculate the metrics by class. Defaults to False.
+            no_class (bool, optional): whether to calculate the metrics between labels and detections while ignoring the class. Defaults to False.
+
+        Returns:
+            list: list of written files
+        """
+
+        logger.info("Loading labels as a GeoPandas DataFrame...")
+        labels_gdf = gpd.read_file(labels_path)
+        labels_gdf = labels_gdf.to_crs(2056)
+        if 'year' in labels_gdf.keys():  
+            labels_gdf['year'] = labels_gdf.year.astype(int)       
+            labels_gdf = labels_gdf.rename(columns={"year": "year_label"})
+        logger.success(f"{DONE_MSG} {len(labels_gdf)} features were found.")
+
+        # get classe ids
+        categories_info_df, id_classes = get_categories(categories_path)
+
+        # append class ids to labels
+        labels_gdf['CATEGORY'] = labels_gdf.CATEGORY.astype(str)
+        labels_w_id_gdf = labels_gdf.merge(categories_info_df, on='CATEGORY', how='left')
+
+        if drop_year:
+            labels_w_id_gdf.drop(columns='year_label', inplace=True)
+            dets_gdf.drop(columns='year_det', inplace=True)
+
+        if no_class:
+            labels_w_id_gdf['label_class'] = 1  # label classes start at 1
+            dets_gdf['det_class'] = 0           # det classes start at 0
+            labels_w_id_gdf['CATEGORY'] = 'human activity'        
+
+        dets_gdf_dict = {}
+        clipped_labels_gdf_dict = {}
+        if KEEP_DATASET_SPLIT:
+            split_tiles_gdf = gpd.read_file(os.path.join(os.path.dirname(labels_path), f'split_aoi_tiles{"_250410" if "_250410" in labels_path else ""}.geojson'))
+            split_tiles_gdf = split_tiles_gdf.to_crs(2056)
+            tiles_gdf = split_tiles_gdf.dissolve(['dataset'] + ([] if drop_year else ['year_tile']), as_index=False)
+
+            clipped_labels_gdf = clip_objects(labels_w_id_gdf, tiles_gdf, fact=0.99999)
+
+            if 'dataset' in dets_gdf.columns:
+                clipped_detections_gdf = dets_gdf.copy()
+            else:
+                logger.warning('No column for the dataset information in the detections. Detections will be clipped to tiles.')
+                clipped_detections_gdf = clip_objects(dets_gdf, tiles_gdf, fact=0.99999)
```

Comment:
clip_objects seemed to be used to add a column, not really for clipping. Change `fact` to 1?